server/home/home-service/src/main/java/io/holoinsight/server/home/biz/plugin/model/HostingAlert.java

```diff
@@ -0,0 +1,245 @@
+
+    return alarmRuleDTO;
+  }
+
+  private String getAlertTypeDesc(String alertType) {
+    String desc = "规则";
+    if ("ai".equalsIgnoreCase(alertType)) {
+      desc = "智能";
+    }
+    return desc;
+  }
+
+  private Map<String, Object> buildTimeFilter() {
+    TimeFilter timeFilter = new TimeFilter();
+    timeFilter.setModel(TimeFilterEnum.DAY.getDesc());
+    timeFilter.setFrom("00:00:00");
+    timeFilter.setTo("23:59:59");
+    timeFilter.setWeeks(Collections.emptyList());
+    return J.toMap(J.toJson(timeFilter));
+  }
+
+  private Map<String, Object> buildRule(IntegrationProductDTO product, List<Filter> filters) {
+    Rule alertRule = new Rule();
+    alertRule.setBoolOperation(BoolOperationEnum.AND);
+    alertRule.setTriggers(buildTriggers(product, filters));
+    return J.toMap(J.toJson(alertRule));
+  }
+
+  private List<Trigger> buildTriggers(IntegrationProductDTO product, List<Filter> filters) {
+    Trigger trigger = new Trigger();
+    trigger.setQuery("a");
+    trigger.setAggregator(slidingWindowAggregator);
+    trigger.setDownsample(
+        StringUtils.isEmpty(slidingWindowSize) ? 1 : Long.valueOf(slidingWindowSize));
+    trigger.setStepNum(StringUtils.isEmpty(stepNum) ? 1 : Integer.valueOf(stepNum));
+    trigger.setType(FunctionEnum.valueOf(functionType));
+    trigger.setTriggerContent(buildTriggerContent());
+    trigger.setDatasources(buildDatasources(filters));
+    trigger.setRuleConfig(this.ruleConfig);
+    trigger.setCompareConfigs(this.compareConfigs);
+    return Arrays.asList(trigger);
+  }
+
+  // sample: CPU使用率 最近5个周期平均值大于等于80 周期为一分钟
+  protected String buildTriggerContent() {
+    StringBuilder content = new StringBuilder();
+    if ("ai".equalsIgnoreCase(this.alertType)) {
+      content.append(this.alertMetricAlias) //
+          .append(this.functionType) //
+          .append("智能告警");
+    } else {
+      content.append(alertMetricAlias) //
+          .append("最近").append(stepNum) //
+          .append("个周期的").append(slidingWindowAggregator) //
+          .append("值").append(parseCompare(compareConfigs)) //
+          .append(", 周期为").append(slidingWindowSize) //
+          .append("分钟");
+    }
+    return content.toString();
+  }
+
+  private String parseCompare(List<CompareConfig> compareConfigs) {
+    if (CollectionUtils.isEmpty(compareConfigs)) {
+      return StringUtils.EMPTY;
+    }
+    List<String> result = new ArrayList<>();
+    for (CompareConfig compareConfig : compareConfigs) {
+      List<CompareParam> compareParams = compareConfig.getCompareParam();
+
+      List<String> contents = new ArrayList<>();
+      for (CompareParam param : compareParams) {
+        contents.add(getCompareDesc(param.getCmp()) + param.getCmpValue());
+      }
+      String condition = String.join(" and ", contents);
+      result.add("[" + condition + "]");
+    }
+    return String.join(" or ", result);
+  }
+
+  private String getCompareDesc(CompareOperationEnum cmp) {
```

Comment:
Can here use English? Or supports i18n?

Reply:
Agree, we should support i18n in the future. It's better to address this in a dedicated PR.